### ISSUE TITLE
NoneType Error Inquiry

### ISSUE THREAD

**Issue by Ashllives** [wwcodemanila/WWCodeManila-Python#33]
I am writing some codes for my project in Coursera and encountered an error: "TypeError: list indices must be integers, not NoneType". I would like to use my variable as part of my condition in a while loop. 

http://www.codeskulptor.org/#user43_8HjJIKeLg9_59.py

P.S. I was trying to check if the generated coordinates [row,col] in my grid already has an existing value or not. Kasi po natatabunan ng new tile ang existing tile sometimes during the game. And excuses po sa magulo kong code. :)

**Comment by dualcnhq** [comment]
Hi @Ashllives, 

I checked your code and there's an error in `Line: 285` in `get_tile` function so I tried to debug it by printing the values as stated below:

```
def get_tile(self, row, col):
        """
        Return the value of the tile at position row, col.
        """
        print self._grid
        print row
        print col
        return self._grid[row][col]
``` 
The result is:

```
random_choice = 0.825177832608
[[0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0]]
None
None
Line 288: TypeError: list indices must be integers, not NoneType
```

I checked all the occurrences of `get_tile` function which brought me to `check_tile` and I checked the value of `row` and `col` again and got `None` result. 
So I check all the occurrences of `check_tile` function which brought me to `new_tile` function, so in `Line 239` you'll see a `while loop` calling the `check_tile` function with params `prev_row`, `prev_col` which have values of `None`

I tried to change the value of `prev_row = 0`, `prev_col = 0` and the `NoneType` error is gone. 

In conclusion, the error occurred because the values you are trying to pass `prev_row` and `prev_col` is `None` which is expected to be an `integer` since you have a list indices of `integers` :smile: 

Hope this helps.
Let us know if you have other question/s...

PS.
While debugging code, it's also helpful if you use `print` to see the values you're trying to return and see if it's returning the desired result you want and you can delete it after you confirmed it. :smile_cat:

**Comment by Ashllives** [comment]
Thanks for all the help Ms Nicole and Ms Clau. I'm done with my project and got 92% out of it. :) Cheers!
